ui/pages/login.tsx

```diff
@@ -297,6 +302,12 @@ const Login: NextPage = () => {
       });
 
       return null;
+    } else if (isWebauthn) {
+      // TODO: Refactor me
+      const u = renderFlow.ui.action.split("self-service");
+      if (u.length == 2) {
+        renderFlow.ui.action = baseURL + "/api/kratos/self-service" + u[1];
+      }
```

Comment:
This probably shouldn't be in the else branch?
```suggestion
    }
    if (isWebauthn) {
      // TODO: Refactor me
      const u = renderFlow.ui.action.split("self-service");
      if (u.length == 2) {
        renderFlow.ui.action = baseURL + "/api/kratos/self-service" + u[1];
      }
```

Reply:
You probably want to [split on first occurrence](https://stackoverflow.com/questions/4607745/split-string-only-on-first-instance-of-specified-character) and not split and hope that you only get one match to drop the `u.length == 2` condition.

Reply:
Also probably best to match "/self-service/", to avoid matching parts of the domain.

Reply:
Not sure how the SDK would help here.

Reply:
The sdk is already configured with the correct URL and will send the payload in JSON instead of FORM. This custom logic wouldnt be required if we used the sdk to make the request.

Reply:
Ah yes, this would also save us from setting up a base url state (see the other comment). Just not sure if we can get the correct url out of it. Because here we need to set the form target url and not issue a request directly.

Reply:
Could we add a listener for the `submit` event to this form so that we can override the default behavior and use the sdk instead?

Reply:
Otherwise I guess we will just have to handle it in the backend.

Reply:
> Could we add a listener for the submit event to this form so that we can override the default behavior and use the sdk instead?

That is possible, but seems a bit hacky.

I think it will be better to construct the base url in the same way we do it for the kratos sdk. We could make that logic reusable. Then constructing the prefixed form target url similar you do it here already.